react/features/jwt/middleware.js

```diff
@@ -0,0 +1,60 @@
+import { SET_ROOM_URL } from '../base/conference';
+import { parseURLParams } from '../base/config';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { setIsGuest, setTokenData } from './actions';
+import TokenData from './TokenData';
+
+/**
+ * Middleware to parse token data upon setting a new room URL.
+ *
+ * @param {Store} store - The Redux store.
+ * @private
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case SET_ROOM_URL:
+        return _setRoomURL(store, next, action);
+    }
+
+    return next(action);
+});
+
+/**
+ * Notifies the feature jwt that the action {@link SET_ROOM_URL} is being
+ * dispatched within a specific Redux {@code store}.
+ *
+ * @param {Store} store - The Redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {Dispatch} next - The Redux dispatch function to dispatch the
+ * specified {@code action} to the specified {@code store}.
+ * @param {Action} action - The Redux action {@code SET_ROOM_URL} which is being
+ * dispatched in the specified {@code store}.
+ * @private
+ * @returns {Object} The new state that is the result of the reduction of the
+ * specified {@code action}.
+ */
+function _setRoomURL(store, next, action) {
+    const {
+        enableUserRolesBasedOnToken
+    } = store.getState()['features/base/config'];
+    const isGuest = !enableUserRolesBasedOnToken;
```

Comment:
Oh, that doesn't seem right! You're dispatching `SET_ROOM_URL` before the `config` is downloaded so `isGuest` may not be correct.